ui-v2/app/mixins/token/with-actions.js

```diff
@@ -0,0 +1,60 @@
+import Mixin from '@ember/object/mixin';
+import WithBlockingActions from 'consul-ui/mixins/with-blocking-actions';
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default Mixin.create(WithBlockingActions, {
+  settings: service('settings'),
+  repo: service('settings'),
+  actions: {
+    use: function(item) {
+      return get(this, 'feedback').execute(() => {
+        return get(this, 'repo')
+          .findBySlug(get(item, 'AccessorID'), this.modelFor('dc').dc.Name)
+          .then(item => {
+            return get(this, 'settings')
+              .persist({
+                token: {
+                  AccessorID: get(item, 'AccessorID'),
+                  SecretID: get(item, 'SecretID'),
+                },
+              })
+              .then(() => {
+                // using is similar to delete in that
+                // if you use from the listing page, stay on the listing page
+                // whereas if you use from the detail page, take me back to the listing page
+                return this.afterDelete(...arguments);
```

Comment:
😄 hehe yeah arrow functions 🤷‍♂️ I try to use them sparingly.

You're right tho, in this case it could be confusing for someone looking at this. The code is kind of a 'pipeline' so `arguments` is always the same `item` anyway, so it doesn't matter what the scope is here. But yeah I'll have another look at this, I might change them back to traditional functions - I think I'm doing this everywhere in the app though so might not be part of this PR.

Reply:
I like arrow functions, but yeah using `arguments` can be tricky - but this is doing what you intended so 👌